### PR TITLE
Added AKS Preview registration, Workload Identity

### PR DESCRIPTION
**Pull request by hieumoscow**
# [Issue-id](https://github.com/aztfmod/terraform-azurerm-caf/issues/ISSUE-ID-GOES-HERE)

## PR Checklist

---

- [ ] I have added example(s) inside the [./examples/] folder
- [ ] I have added the example(s) to the integration test list for [normal (~30 minutes)](./workflows/standalone-scenarios.json) or [long runner >30 minutes](./workflows/standalone-scenarios-longrunners.json)
- [ ] I have checked the [coding conventions as per the wiki](https://github.com/aztfmod/terraform-azurerm-caf/wiki)
- [ ] I have checked to ensure there aren't other open Pull Requests for the same update/change?

## Does this introduce a breaking change

- [ ] YES
- [ ] NO

### CODE REVIEW

**Review by arnaudlh — APPROVED** [review]
LGTM

File: preview_features.tf [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+resource "null_resource" "register_feature_preview" {
+  for_each = { for f in local.flat_preview_features : f.full_name => f }
+
+  provisioner "local-exec" {
+    command = <<EOT
+      az feature register --namespace ${each.value.namespace} -n ${each.value.feature}
+      while true; do
+        feature_status=$(az feature show --namespace ${each.value.namespace} -n ${each.value.feature} --query "properties.state" -o tsv)
+        if [ "$feature_status" == "Registered" ]; then
```
Comment (arnaudlh) [inline preview_features.tf:30]:
```suggestion
        if [ -n "$feature_status" ] && [ "$feature_status" = "Registered" ]; then
```

**Review by arnaudlh — CHANGES_REQUESTED** [review]
LGTM, please review change in the bash string processing.

**Review by LaurentLesle — APPROVED** [review]
LGTM.  More generic to support other preview features.

**Review by arnaudlh — APPROVED** [review]
Good

**Pull request merged**